goblet/resources/pubsub.py

```diff
@@ -150,23 +151,44 @@ def _deploy_subscription(self, topic_name, topic, config={}):
             req_body=req_body,
         )
 
-    def _deploy_trigger(self, topic_name, sourceUrl=None, entrypoint=None):
+    def _deploy_trigger(self, topic_name, source=None, entrypoint=None):
```

Comment:
instead of source, lets call it backend since we are passing that class. and lets add an attribute for sourceUrl in the base class that each backend sets. 
so it will be backend.sourceUrl

Reply:
So in v1,  generateuploadurl only outputs sourceUrl, but in v2 we need to pass through the full output of generateUploadUrl (the url and storageSource), so I changed the argument to source.

Reply:
Should I change it to pass through the backend class instead?

Reply:
ah source is just the storage json blob. 
now that i think about it though passing the backend class could makes sense. since we can standardize the sourceurl + other features from the class i think. this should make it easier to handle multiple backend types in each resource.